src/rulesets-new/json-api-rules/__tests__/resource-object-rules.test.ts

```diff
@@ -0,0 +1,312 @@
+                          },
+
+                          links: {
+                            properties: {
+                              self: {
+                                type: "string",
+                              },
+                            },
+                          },
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      } as OpenAPIV3.Document;
+
+      const ruleRunner = new RuleRunner([resourceObjectRules]);
+      const ruleInputs = {
+        ...TestHelpers.createRuleInputs(baseJson, afterJson),
+        context,
+      };
+      const results = ruleRunner.runRulesWithFacts(ruleInputs);
+      expect(results.length).toBeGreaterThan(0);
+      expect(results.every((result) => result.passed)).toBe(true);
+      expect(results).toMatchSnapshot();
+    });
+  });
+
+  describe("valid patch shapes", () => {
+    test("status code 200", () => {
+      const afterJson = {
+        ...baseJson,
+        paths: {
+          "/api/example": {
+            patch: {
+              responses: {
+                "200": {
+                  description: "",
+                  content: {
+                    "application/vnd.api+json": {
+                      schema: {
+                        type: "object",
+                        properties: {
+                          data: {
+                            type: "object",
+                            properties: {
+                              id: {
+                                type: "string",
+                                format: "uuid",
+                              },
+                              type: {
+                                type: "string",
+                              },
+                            },
+                          },
+                          jsonapi: {
+                            type: "string",
+                          },
+                          links: {
+                            properties: {
+                              self: {
+                                type: "string",
+                              },
+                            },
+                          },
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      } as OpenAPIV3.Document;
+
+      const ruleRunner = new RuleRunner([resourceObjectRules]);
+      const ruleInputs = {
+        ...TestHelpers.createRuleInputs(baseJson, afterJson),
+        context,
+      };
+      const results = ruleRunner.runRulesWithFacts(ruleInputs);
+      expect(results.length).toBeGreaterThan(0);
+      expect(results.every((result) => result.passed)).toBe(true);
+      expect(results).toMatchSnapshot();
+    });
+  });
+
+  describe("valid delete shapes", () => {
+    test("status code 200", () => {
```

Comment:
I prefer the "passes when ..." / "fails when ..." test naming we had in the property rules, because you could have variations on this like:

"passes when a delete responds with 200 and there is a JSON API response body"
"fails when a delete responds with 200 and there is no response body"
"passes when a delete responds with 204 and there is no response body"
"fails when a delete responds with 204 and there is a response body"

Reply:
Makes sense! Updated test cases names